File: Controllers/CreateMauiAppController.cs


namespace MauiApp.Controllers
{
    using Microsoft.AspNetCore.Mvc;
    using MauiApp.Services;

    [ApiController]
    [Route("api/[controller]")]
    public class MauiAppController : ControllerBase
    {
        private readonly IMauiProgram _mauiProgram;

        public MauiAppController(IMauiProgram mauiProgram)
        {
            _mauiProgram = mauiProgram;
        }

        [HttpPost]
        public IActionResult CreateMauiApplication(string parameter)
        {
            var mauiApp = _mauiProgram.CreateMauiApp(parameter);
            return Ok(mauiApp);
        }
    }
}

File: src/app/controllers/CreateMauiAppController.ts


import { MauiProgramService } from '../services/MauiProgramService';

class CreateMauiAppController {
  private mauiProgramService: MauiProgramService;

  constructor(mauiProgramService: MauiProgramService) {
    this.mauiProgramService = mauiProgramService;
  }

  async createMauiApp() {
    return await this.mauiProgramService.createMauiApp();
  }
}

export default CreateMauiAppController;